be/src/exec/aggregator.h

```diff
@@ -136,6 +136,30 @@ enum AggrMode {
     AM_STREAMING_POST_CACHE
 };
 
+enum AggrAutoState { INIT_PREAGG = 0, ADJUST, PASS_THROUGH, FORCE_PREAGG, PREAGG, SELECTIVE_PREAGG };
+
+struct AggrAutoContext {
```

Comment:
we may add a new file? aggregator.cpp is too large?

Reply:
this file has 527 lines including these codes, I think this is not too large.